chainstate/test-suite/src/tests/nft_issuance.rs

```diff
@@ -0,0 +1,1288 @@
+
+            let block = tf.block(block_id);
+            let outputs = &TestBlockInfo::from_block(&block).txns[0].1;
+            let issuance_output = &outputs[0];
+
+            match issuance_output.value().token_data().unwrap() {
+                TokenData::NftIssuanceV1(nft) => {
+                    assert_eq!(nft.metadata.media_hash(), &media_hash);
+                }
+                _ => panic!("NFT issuance not found"),
+            }
+        }
+    })
+}
+
+#[rstest]
+#[trace]
+#[case(Seed::from_entropy())]
+fn nft_issuance_valid_case(#[case] seed: Seed) {
+    utils::concurrency::model(move || {
+        let mut tf = TestFramework::default();
+        let outpoint_source_id = TestBlockInfo::from_genesis(&tf.genesis()).txns[0].0.clone();
+        let mut rng = make_seedable_rng(seed);
+
+        let max_desc_len = tf.chainstate.get_chain_config().token_max_description_len();
+        let max_name_len = tf.chainstate.get_chain_config().token_max_name_len();
+        let max_ticker_len = tf.chainstate.get_chain_config().token_max_ticker_len();
+        let valid_rfc3986_uri =
+            b"https://something.com/?a:b.c_d-e~f!g/h?I#J[K]L@M$N&O/P'Q(R)S*T+U,V;W=Xyz".to_vec();
+
+        let output_value = NftIssuanceV1 {
+            metadata: Metadata {
+                creator: random_creator(),
+                name: random_string(&mut rng, 1..max_name_len).into_bytes(),
+                description: random_string(&mut rng, 1..max_desc_len).into_bytes(),
+                ticker: random_string(&mut rng, 1..max_ticker_len).into_bytes(),
+                icon_uri: DataOrNoVec::from(Some(valid_rfc3986_uri.clone())),
+                additional_metadata_uri: DataOrNoVec::from(Some(valid_rfc3986_uri.clone())),
+                media_uri: DataOrNoVec::from(Some(valid_rfc3986_uri)),
+                media_hash: vec![1, 2, 3, 4, 5, 6, 7, 8, 9, 0],
+            },
+        };
+
+        let block_index = tf
+            .make_block_builder()
+            .add_transaction(
+                TransactionBuilder::new()
+                    .add_input(
+                        TxInput::new(outpoint_source_id, 0),
+                        InputWitness::NoSignature(None),
+                    )
+                    .add_output(TxOutput::new(
+                        output_value.clone().into(),
+                        OutputPurpose::Transfer(Destination::AnyoneCanSpend),
+                    ))
+                    .build(),
+            )
+            .build_and_process()
+            .unwrap()
+            .unwrap();
+
+        assert_eq!(
+            block_index.block_height(),
+            common::primitives::BlockHeight::from(1)
+        );
+
+        let block = tf.block(*block_index.block_id());
+        let outputs = &TestBlockInfo::from_block(&block).txns[0].1;
+        let issuance_output = &outputs[0];
+
+        assert_eq!(issuance_output.value(), &output_value.into());
+    })
+}
```

Comment:
You don't seem to have tests for media hash length. Since you're enforcing that, you should test it.

Reply:
So, I have `nft_media_hash_valid`, `nft_media_hash_too_long`, `nft_media_hash_too_short`. Shall I add checks in valid case?